packages/store/src/symbols.ts

```diff
@@ -79,7 +79,24 @@ export class NgxsConfig {
   }
 }
 
-export type StateOperator<T> = (existing: Readonly<T>) => T;
+type _NoInfer<T> = T extends infer S ? S : never;
+export type NoInfer<T> = T extends (infer O)[] ? _NoInfer<O>[] : _NoInfer<T>;
```

Comment:
Since we're exposing it publicly could you leave meaningful comments?

Reply:
@arturovt I have added the doc comments now.